app/models/user.rb

```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :username, presence: true, length: { in: 6..50 }
+  validates :email, presence: true, format: { with: URI::MailTo::EMAIL_REGEXP }
+  validates :password, presence: true, confirmation: true, length: { minimum: 8 }
+  validates :password_confirmation, presence: true
+end
```

Comment:
It is a good idea not to repeat the code, the validation for the password has been repeated. You can remove the last line.

Reply:
No, I can't: password confirmation skips when `:password_confirmation` is nil, so you need to validate its presence. See the [docs](https://guides.rubyonrails.org/active_record_validations.html#confirmation)

Reply:
Okay, if that's the case, then just make the other change and submit